internal/collector/processor/nodebatcher/node_batcher.go

```diff
@@ -421,6 +415,7 @@ func (b *batch) add(spans []*tracepb.Span) (cut bool, closed bool) {
 		b.grow(openTill)
 	}
 
+	// Why not: copy(b.items.Load().([]*tracepb.Span)[openFrom:openTill], spans)?
```

Comment:
@bogdandrutu we should just supplant this all this logic with something simpler as you said. https://github.com/census-instrumentation/opencensus-service/pull/487 replaces this, addressing the PR comments there now

Reply:
As I mentioned in my previous comment I feel like I am changing too much in this PR and I want to go with that proposal of still passing the source name for the moment in the TraceData then remove it later from there.